internal/identity/oidc/gitlab/gitlab.go

```diff
@@ -0,0 +1,97 @@
+// Package gitlab implements OpenID Connect for Gitlab
+//
+// https://www.pomerium.io/docs/identity-providers/gitlab.html
+package gitlab
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/coreos/go-oidc"
+	"github.com/pomerium/pomerium/internal/httputil"
+	"github.com/pomerium/pomerium/internal/identity/oauth"
+	pom_oidc "github.com/pomerium/pomerium/internal/identity/oidc"
+	"github.com/pomerium/pomerium/internal/log"
+	"github.com/pomerium/pomerium/internal/sessions"
+	"github.com/pomerium/pomerium/internal/version"
+)
+
+// Name identifies the GitLab identity provider
+const Name = "gitlab"
+
+var defaultScopes = []string{oidc.ScopeOpenID, "api", "read_user", "profile", "email"}
```

Comment:
update: we can change the `api` scope to `read_api` now based on the new release from gitlab
https://docs.gitlab.com/ee/user/profile/personal_access_tokens.html#limiting-scopes-of-a-personal-access-token

Reply:
👍 let's fix in a FUP pr so we can reference https://gitlab.com/gitlab-org/gitlab/-/merge_requests/28944

Reply:
That's cool!